### DISCUSSION TITLE
ESPROM Program Memory

### DISCUSSION THREAD

**Discussion by MnaBad** [hneemann/Digital#1239] [Q&A] [answered]
I'm making a CPU and need an ESPROM to store my program. I am using an external compiler which gives me hex data to put right into the ESPROM. I heard you can just paste the program into a file after ticking the Program Memory box in ESPROM advanced settings but I don't see any file with the .hex extension.

**Comment by hneemann** [comment, marked as answer]
I'm not quite sure what your question is. If the "Program memory" checkbox is set in a memory component, a file name can be selected in the circuit-specific settings. When the simulation is started, the file content is then loaded into the memory.
The old Logisim hex files, Intel hex files or binary files are supported.

**Reply by MnaBad** [reply]
Thanks, the explanation inside the documentation is pretty vague when it comes to the program memory. I didn't know that you had to select a specific file in circuit-specific-settings, I thought it generated one for you automatically.
